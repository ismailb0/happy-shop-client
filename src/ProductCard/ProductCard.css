.product-card {
  display: flex;
  flex-direction: column;
  margin: 0 1vw;
}

.card-media {
  height: 40vh;
}

.card-content {
  font-size: smaller;
  height: 14vh;
  padding: 1vh 1vw;
  text-align: left;
  width: 18vw;
}

.product-brand {
  font-weight: bold;
  margin: 7px 0px;
}

.product-price {
  font-weight: bold;
  margin: 7px 0px;
}
