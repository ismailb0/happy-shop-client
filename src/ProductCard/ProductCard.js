import React, { Component } from 'react';

import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';

import './ProductCard.css';

class ProductCard extends Component {

  render() {
    return (
      <Card className='product-card'>
        <CardMedia
          className="card-media"
          image={this.props.image}
          title={this.props.name}
        />
        <div className="card-content">
          <div className='product-brand'>
            {this.props.brand}
          </div>
          <div>
            {this.props.name}
          </div>
          <div className='product-price'>
            {`$${this.props.price}`}
          </div>
        </div>
      </Card>
    );
  }
}

export default ProductCard;
