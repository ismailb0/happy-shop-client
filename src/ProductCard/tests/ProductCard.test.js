import React from 'react';
import { shallow } from "enzyme";
import ProductCard from '../ProductCard';

describe('<ProductCard />', () => {

  const props = {
    image: 'testImage.jpg',
    name: 'testName',
    brand: 'testBrand',
    price: 59.82,
  }

  it('should match snapshot', () => {
    const renderedComponent = shallow(<ProductCard {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

});
