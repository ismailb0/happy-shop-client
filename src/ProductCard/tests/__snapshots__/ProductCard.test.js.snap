// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductCard
    brand="testBrand"
    image="testImage.jpg"
    name="testName"
    price={59.82}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(CardMedia)
          className="card-media"
          image="testImage.jpg"
          title="testName"
        />,
        <div
          className="card-content"
        >
          <div
            className="product-brand"
          >
            testBrand
          </div>
          <div>
            testName
          </div>
          <div
            className="product-price"
          >
            $59.82
          </div>
        </div>,
      ],
      "className": "product-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card-media",
          "image": "testImage.jpg",
          "title": "testName",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="product-brand"
            >
              testBrand
            </div>,
            <div>
              testName
            </div>,
            <div
              className="product-price"
            >
              $59.82
            </div>,
          ],
          "className": "card-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "testBrand",
              "className": "product-brand",
            },
            "ref": null,
            "rendered": "testBrand",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "testName",
            },
            "ref": null,
            "rendered": "testName",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "$59.82",
              "className": "product-price",
            },
            "ref": null,
            "rendered": "$59.82",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardMedia)
            className="card-media"
            image="testImage.jpg"
            title="testName"
          />,
          <div
            className="card-content"
          >
            <div
              className="product-brand"
            >
              testBrand
            </div>
            <div>
              testName
            </div>
            <div
              className="product-price"
            >
              $59.82
            </div>
          </div>,
        ],
        "className": "product-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card-media",
            "image": "testImage.jpg",
            "title": "testName",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="product-brand"
              >
                testBrand
              </div>,
              <div>
                testName
              </div>,
              <div
                className="product-price"
              >
                $59.82
              </div>,
            ],
            "className": "card-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "testBrand",
                "className": "product-brand",
              },
              "ref": null,
              "rendered": "testBrand",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "testName",
              },
              "ref": null,
              "rendered": "testName",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$59.82",
                "className": "product-price",
              },
              "ref": null,
              "rendered": "$59.82",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
